khal/khalendar/event.py

```diff
@@ -919,110 +933,90 @@ def duration(self) -> dt.timedelta:
 
 
 def create_timezone(
-    tz: pytz.BaseTzInfo,
-    first_date: Optional[dt.datetime]=None,
-    last_date: Optional[dt.datetime]=None
+    tz,
+    first_date: Optional[dt.datetime] = None,
+    last_date: Optional[dt.datetime] = None,
 ) -> icalendar.Timezone:
     """
-    create an icalendar vtimezone from a pytz.tzinfo object
-
-    :param tz: the timezone
-    :param first_date: the very first datetime that needs to be included in the
-    transition times, typically the DTSTART value of the (first recurring)
-    event
-    :param last_date: the last datetime that needs to included, typically the
-    end of the (very last) event (of a recursion set)
-    :returns: timezone information
-
-    we currently have a problem here:
-
-       pytz.timezones only carry the absolute dates of time zone transitions,
-       not their RRULEs. This will a) make for rather bloated VTIMEZONE
-       components, especially for long recurring events, b) we'll need to
-       specify for which time range this VTIMEZONE should be generated and c)
-       will not be valid for recurring events that go into eternity.
-
-    Possible Solutions:
-
-    As this information is not provided by pytz at all, there is no
-    easy solution, we'd really need to ship another version of the OLSON DB.
-
+    Create an icalendar vtimezone from a pytz.tzinfo or zoneinfo.ZoneInfo object.
     """
-    if isinstance(tz, StaticTzInfo):
+    if isinstance(tz, pytz.tzinfo.StaticTzInfo):
         return _create_timezone_static(tz)
 
-    # TODO last_date = None, recurring to infinity
+    first_date = dt.datetime.today() if not first_date else first_date
+    last_date = first_date + dt.timedelta(days=1) if not last_date else last_date
```

Comment:
I'm not entirely familiar with this code. Why did you remove `to_naive_utc` here?